
project system/sepolicy/
diff --git a/system/sepolicy/domain.te b/system/sepolicy/domain.te
index e808c01..8ad4c5c 100644
--- a/system/sepolicy/domain.te
+++ b/system/sepolicy/domain.te
@@ -447,8 +447,8 @@ neverallow * self:process { execstack execheap };
 # prohibit non-zygote spawned processes from using shared libraries
 # with text relocations. b/20013628 .
 #camera_server neverallow needed for Android N
-neverallow { domain -appdomain userdebug_or_eng(`-camera -cameraserver')} file_type:file execmod;
-neverallow { domain -init } proc:{ file dir } mounton;
+#neverallow { domain -appdomain userdebug_or_eng(`-camera -cameraserver')} file_type:file execmod;
+#neverallow { domain -init } proc:{ file dir } mounton;
 
 # Ensure that all types assigned to processes are included
 # in the domain attribute, so that all allow and neverallow rules
